datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/timeline/mappers/SchemaFieldBlameMapper.java

```diff
@@ -0,0 +1,188 @@
+          new SemanticVersionStruct(truncateSemanticVersion(changeTransaction.getSemVer()),
+              changeTransaction.getTimestamp(), changeTransaction.getVersionStamp());
+      semanticVersionStructList.add(semanticVersionStruct);
+
+      for (ChangeEvent changeEvent : changeTransaction.getChangeEvents()) {
+        if (changeEvent.getCategory() != ChangeCategory.TECHNICAL_SCHEMA) {
+          continue;
+        }
+
+        String schemaUrn = changeEvent.getModifier();
+        if (schemaUrn == null || schemaBlameMap.containsKey(schemaUrn)) {
+          continue;
+        }
+
+        SchemaFieldBlame schemaFieldBlame = new SchemaFieldBlame();
+
+        SchemaFieldKey schemaFieldKey;
+        try {
+          schemaFieldKey = (SchemaFieldKey) EntityKeyUtils.convertUrnToEntityKey(Urn.createFromString(schemaUrn),
+              new SchemaFieldKey().schema());
+        } catch (Exception e) {
+          log.debug(String.format("Could not generate schema urn for %s", schemaUrn));
+          continue;
+        }
+
+        String fieldPath = schemaFieldKey.getFieldPath();
+        schemaFieldBlame.setFieldPath(fieldPath);
+
+        SchemaFieldChange schemaFieldChange =
+            getLastSchemaFieldChange(changeEvent, changeTransaction.getTimestamp(), changeTransaction.getActor(),
+                changeTransaction.getSemVer(), changeTransaction.getVersionStamp());
+        schemaFieldBlame.setSchemaFieldChange(schemaFieldChange);
+
+        schemaBlameMap.put(schemaUrn, schemaFieldBlame);
+      }
+    }
+
+    result.setSchemaFieldBlameList(schemaBlameMap.values()
+        .stream()
+        .filter(schemaFieldBlame -> !schemaFieldBlame.getSchemaFieldChange()
+            .getChangeType()
+            .equals(ChangeOperationType.REMOVE))
+        .collect(Collectors.toList()));
+    result.setSemanticVersionList(semanticVersionStructList);
+    return result;
+  }
+
+  private static Optional<Pair<SemanticVersion, ChangeTransaction>> semanticVersionChangeTransactionPair(
+      ChangeTransaction changeTransaction) {
+    Optional<SemanticVersion> semanticVersion = createSemanticVersion(changeTransaction.getSemVer());
+    return semanticVersion.map(version -> Pair.of(version, changeTransaction));
+  }
+
+  private static Optional<SemanticVersion> createSemanticVersion(String semanticVersionString) {
+    String truncatedSemanticVersion = truncateSemanticVersion(semanticVersionString);
+    try {
+      SemanticVersion semanticVersion = SemanticVersion.parse(truncatedSemanticVersion);
+      return Optional.of(semanticVersion);
+    } catch (SemanticVersion.SemanticVersionParseException e) {
+      return Optional.empty();
+    }
+  }
+
+  // The SemanticVersion is currently returned from the ChangeTransactions in the format "x.y.z-computed". This function
+  // removes the suffix "computed".
+  private static String truncateSemanticVersion(String semanticVersion) {
+    String suffix = "-computed";
```

Comment:
we might want to put this in the timeline api. feels a bit weird to return -computed cc @RyanHolstien for later.

Reply:
We had this in to make it clear to users that these were not set in stone versions and a version could be different given a different request with the intent that eventually they would be persisted. We can take it off, but it needs to be very clear to users that these are calculated values.

Reply:
Agreeed

Reply:
but by putting in the version identifier itself.. feels like not the "ideal" place